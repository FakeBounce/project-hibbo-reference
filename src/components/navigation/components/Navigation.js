import React from 'react';
import { shape, oneOf, string, func, arrayOf } from 'prop-types';
import { View, TouchableOpacity, Text } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import theme from 'styles/appStyles';
import { getTranslations } from 'utils/i18n';

import MenuLink from '../../MenuLink';

const styles = EStyleSheet.create({
  background: {
    flex: 1,
    backgroundColor: '#fffe82',
  },
  content: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
    marginTop: 130.5,
  },
  text: {
    color: '#303030',
    fontSize: 18,
    marginBottom: 25,
    fontFamily: '$fonts.circularBook',
  },
  textIpad: {
    color: '#303030',
    fontSize: 28,
    marginBottom: 25,
    fontFamily: '$fonts.circularBook',
  },
  button: { alignSelf: 'center' },
});

const Navigation = ({ mode, navigation, list, goTo }) => {
  return (
    <View style={[styles.background, theme.cardStyle]}>
      <View style={styles.content}>
        {list.map(link => {
          return (
            <MenuLink
              style={[styles.link, link.style]}
              Textstyle={link.Textstyle}
              key={`${link.entitled}`}
              to={() => link.linkto(navigation, goTo)}
              text={link.entitled}
            />
          );
        })}
      </View>
      {mode !== 'children' && (
        <View>
          <TouchableOpacity style={styles.button} onPress={() => {}}>
            <Text style={styles.text}>
              {getTranslations('navigation.myMonimalz')}
            </Text>
          </TouchableOpacity>
        </View>
      )}
    </View>
  );
};

Navigation.defaultProps = {
  list: [
    {
      entitled: getTranslations('navigation.transfert'),
      linkto: (navigation, goTo) => {
        goTo(navigation, 'TransfertStack', { tab: 0 });
      },
    },
    {
      entitled: getTranslations('navigation.moneyPot'),
      linkto: (navigation, goTo) => {
        goTo(navigation, 'TransfertStack', { tab: 2 });
      },
    },
    {
      entitled: getTranslations('navigation.goals'),
      linkto: navigation => {
        navigation.navigate('Mission');
      },
    },
    {
      entitled: getTranslations('navigation.messages'),
      linkto: navigation => {
        navigation.navigate('Notification');
      },
    },
    {
      entitled: getTranslations('navigation.shop'),
      linkto: () => {},
    },
    {
      entitled: getTranslations('navigation.settings'),
      linkto: navigation => {
        navigation.navigate('Settings');
      },
    },
  ],
};

Navigation.propTypes = {
  goTo: func.isRequired,
  mode: oneOf(['children', 'adult']).isRequired,
  list: arrayOf(
    shape({
      entitled: string.isRequired,
      linkto: func.isRequired,
    }),
  ),
  navigation: shape({
    dispatch: func.isRequired,
    goBack: func.isRequired,
    navigate: func.isRequired,
    setParams: func.isRequired,
    state: shape({
      key: string.isRequired,
      routeName: string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default Navigation;
