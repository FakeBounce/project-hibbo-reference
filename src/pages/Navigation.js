import React from 'react';
import NavigationComponent from '../components/navigation';

const Navigation = props => {
  return (
    <NavigationComponent {...props} />
  );
};

export default Navigation;
