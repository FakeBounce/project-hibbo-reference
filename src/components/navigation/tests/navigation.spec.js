import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Navigation from '../components/Navigation';
import NavigationContainer from '../';

describe('Navigation', () => {
  const nav = {
    dispatch: () => {},
    goBack: () => {},
    navigate: () => {},
    setParams: () => {},
    state: {
      key: '',
      routeName: '',
    },
  };
  const mockStore = configureStore();
  const store = mockStore({
    device: 'iphone',
  });


  it('should render the container', () => {
    const navigation = renderer.create(
      <Provider store={store}>
        <NavigationContainer navigation={nav} goTo={jest.fn} />
      </Provider>,
    ).toJSON();

    expect(navigation).toMatchSnapshot();
  });

  it('should render the component for adult', () => {
    const navigation = renderer.create(
      <Provider store={store}>
        <Navigation navigation={nav} mode="adult" goTo={jest.fn} />
      </Provider>,
    ).toJSON();

    expect(navigation).toMatchSnapshot();
  });

  it('should render the component for children', () => {
    const navigation = renderer.create(
      <Provider store={store}>
        <Navigation navigation={nav} mode="children" goTo={jest.fn} />
      </Provider>,
    ).toJSON();

    expect(navigation).toMatchSnapshot();
  });
});
