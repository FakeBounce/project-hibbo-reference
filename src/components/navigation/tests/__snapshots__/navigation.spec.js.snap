// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render the component for adult 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "borderRadius": 10,
        "shadowColor": "transparent",
        "shadowOpacity": 0,
      },
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Transfer
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Money Pot
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Goals
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Messages
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Shop
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        My Monimalz
      </Text>
    </View>
  </View>
</View>
`;

exports[`Navigation should render the component for children 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "borderRadius": 10,
        "shadowColor": "transparent",
        "shadowOpacity": 0,
      },
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Transfer
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Money Pot
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Goals
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Messages
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Shop
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Navigation should render the container 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "borderRadius": 10,
        "shadowColor": "transparent",
        "shadowOpacity": 0,
      },
    ]
  }
>
  <View
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Transfer
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Money Pot
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Goals
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Messages
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Shop
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <View>
          <View
            collapsable={undefined}
            style={
              Object {
                "height": 62,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
          ]
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "#303030",
              "borderRadius": 70,
              "height": 70,
              "opacity": 0,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        My Monimalz
      </Text>
    </View>
  </View>
</View>
`;
